[
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/630446826",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-630446826",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 630446826,
        "node_id": "MDEyOklzc3VlQ29tbWVudDYzMDQ0NjgyNg==",
        "user": {
            "login": "SantaDiver",
            "id": 17476424,
            "node_id": "MDQ6VXNlcjE3NDc2NDI0",
            "avatar_url": "https://avatars.githubusercontent.com/u/17476424?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/SantaDiver",
            "html_url": "https://github.com/SantaDiver",
            "followers_url": "https://api.github.com/users/SantaDiver/followers",
            "following_url": "https://api.github.com/users/SantaDiver/following{/other_user}",
            "gists_url": "https://api.github.com/users/SantaDiver/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/SantaDiver/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/SantaDiver/subscriptions",
            "organizations_url": "https://api.github.com/users/SantaDiver/orgs",
            "repos_url": "https://api.github.com/users/SantaDiver/repos",
            "events_url": "https://api.github.com/users/SantaDiver/events{/privacy}",
            "received_events_url": "https://api.github.com/users/SantaDiver/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-05-18T21:33:01Z",
        "updated_at": "2020-05-18T21:38:04Z",
        "author_association": "NONE",
        "body": "I just downgraded Pytroch to 1.4.0 and this error has gone. Looks like this problem only affects newer version.\r\n\r\nAnd now I face another strange behavior: ONNX returns all \"nan\" as prediction on both optimized and raw models. And this new problem appears only when I'm using CUDAExecutionProvider. With CPUExecutionProvider model returns reasonable numbers.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/630446826/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/631006401",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-631006401",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 631006401,
        "node_id": "MDEyOklzc3VlQ29tbWVudDYzMTAwNjQwMQ==",
        "user": {
            "login": "faxu",
            "id": 20780999,
            "node_id": "MDQ6VXNlcjIwNzgwOTk5",
            "avatar_url": "https://avatars.githubusercontent.com/u/20780999?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/faxu",
            "html_url": "https://github.com/faxu",
            "followers_url": "https://api.github.com/users/faxu/followers",
            "following_url": "https://api.github.com/users/faxu/following{/other_user}",
            "gists_url": "https://api.github.com/users/faxu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/faxu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/faxu/subscriptions",
            "organizations_url": "https://api.github.com/users/faxu/orgs",
            "repos_url": "https://api.github.com/users/faxu/repos",
            "events_url": "https://api.github.com/users/faxu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/faxu/received_events",
            "type": "User",
            "site_admin": true
        },
        "created_at": "2020-05-19T18:39:17Z",
        "updated_at": "2020-05-19T18:39:17Z",
        "author_association": "MEMBER",
        "body": "@spandantiwari can you take a look at the PyTorch issue in the newer version? Can you share the converted onnx model you see the error with?",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/631006401/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/631402420",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-631402420",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 631402420,
        "node_id": "MDEyOklzc3VlQ29tbWVudDYzMTQwMjQyMA==",
        "user": {
            "login": "SantaDiver",
            "id": 17476424,
            "node_id": "MDQ6VXNlcjE3NDc2NDI0",
            "avatar_url": "https://avatars.githubusercontent.com/u/17476424?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/SantaDiver",
            "html_url": "https://github.com/SantaDiver",
            "followers_url": "https://api.github.com/users/SantaDiver/followers",
            "following_url": "https://api.github.com/users/SantaDiver/following{/other_user}",
            "gists_url": "https://api.github.com/users/SantaDiver/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/SantaDiver/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/SantaDiver/subscriptions",
            "organizations_url": "https://api.github.com/users/SantaDiver/orgs",
            "repos_url": "https://api.github.com/users/SantaDiver/repos",
            "events_url": "https://api.github.com/users/SantaDiver/events{/privacy}",
            "received_events_url": "https://api.github.com/users/SantaDiver/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-05-20T10:56:23Z",
        "updated_at": "2020-05-20T13:08:04Z",
        "author_association": "NONE",
        "body": "@faxu Here are my models: [raw](https://drive.google.com/open?id=1krber8QWZX_6VTmyhtbiM2ftkTtPYk9D), [gpu optimized](https://drive.google.com/open?id=1-605zZ38f1YgAz9Erxf_wasQNAsDSB_a).\r\n\r\nUPD: the problem only reproduces with pytorch==1.5.0, onnxruntime==1.2.0, onnxruntime-gpu==1.2.0\r\nNewer onnxruntime==1.3.0, onnxruntime-gpu==1.3.0 fixed it.\r\n\r\nAnd about nans. I have multiple CUDA installations on my machine. Maybe ONNX Runtime uses the wrong one therefore producing nans? I have cudatoolkit==10.1.243 installed in my conda environment.\r\n\r\nUPD: I just  completely removed all CUDA installations and installed CUDA 10.1. NaNs problem is still here.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/631402420/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/639020842",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-639020842",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 639020842,
        "node_id": "MDEyOklzc3VlQ29tbWVudDYzOTAyMDg0Mg==",
        "user": {
            "login": "tianleiwu",
            "id": 30328909,
            "node_id": "MDQ6VXNlcjMwMzI4OTA5",
            "avatar_url": "https://avatars.githubusercontent.com/u/30328909?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/tianleiwu",
            "html_url": "https://github.com/tianleiwu",
            "followers_url": "https://api.github.com/users/tianleiwu/followers",
            "following_url": "https://api.github.com/users/tianleiwu/following{/other_user}",
            "gists_url": "https://api.github.com/users/tianleiwu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/tianleiwu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/tianleiwu/subscriptions",
            "organizations_url": "https://api.github.com/users/tianleiwu/orgs",
            "repos_url": "https://api.github.com/users/tianleiwu/repos",
            "events_url": "https://api.github.com/users/tianleiwu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/tianleiwu/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-06-04T18:15:18Z",
        "updated_at": "2020-06-04T18:15:18Z",
        "author_association": "MEMBER",
        "body": "@SantaDiver, \r\n\r\nI noticed that your model has switched the input naming of attention_mask and token_type_ids, so token_type_ids (all zeros) are used as attention mask. Then the tensor became all zeros (since mask are all zero), that causes NaNs in softmax (Hari has a fix for this). \r\n\r\nYou can follow our benchmkark script (https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/transformers/benchmark.py) to export bert model to ONNX.\r\n\r\nTransformers has fixed a similar issue 3 days ago in onnx export: https://github.com/huggingface/transformers/blob/master/src/transformers/convert_graph_to_onnx.py\r\nYou can have a try as well.\r\n\r\nThe epsilon attribute of SkipLayerNormalization is added in onnxruntime 1.3.0. Note that onnxruntime-tools version is 1.3.0.*, and the optimized model might not be backward compatible with older version (1.2.0).",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/639020842/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/644919464",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-644919464",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 644919464,
        "node_id": "MDEyOklzc3VlQ29tbWVudDY0NDkxOTQ2NA==",
        "user": {
            "login": "SantaDiver",
            "id": 17476424,
            "node_id": "MDQ6VXNlcjE3NDc2NDI0",
            "avatar_url": "https://avatars.githubusercontent.com/u/17476424?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/SantaDiver",
            "html_url": "https://github.com/SantaDiver",
            "followers_url": "https://api.github.com/users/SantaDiver/followers",
            "following_url": "https://api.github.com/users/SantaDiver/following{/other_user}",
            "gists_url": "https://api.github.com/users/SantaDiver/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/SantaDiver/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/SantaDiver/subscriptions",
            "organizations_url": "https://api.github.com/users/SantaDiver/orgs",
            "repos_url": "https://api.github.com/users/SantaDiver/repos",
            "events_url": "https://api.github.com/users/SantaDiver/events{/privacy}",
            "received_events_url": "https://api.github.com/users/SantaDiver/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-06-16T17:59:02Z",
        "updated_at": "2020-06-16T17:59:02Z",
        "author_association": "NONE",
        "body": "@tianleiwu,\r\n\r\nthank you for the explanation! And sorry for not answering for so long.\r\nI'll try to convert my model in proper way and write back if I'll face problems.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/644919464/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/645818227",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-645818227",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 645818227,
        "node_id": "MDEyOklzc3VlQ29tbWVudDY0NTgxODIyNw==",
        "user": {
            "login": "tianleiwu",
            "id": 30328909,
            "node_id": "MDQ6VXNlcjMwMzI4OTA5",
            "avatar_url": "https://avatars.githubusercontent.com/u/30328909?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/tianleiwu",
            "html_url": "https://github.com/tianleiwu",
            "followers_url": "https://api.github.com/users/tianleiwu/followers",
            "following_url": "https://api.github.com/users/tianleiwu/following{/other_user}",
            "gists_url": "https://api.github.com/users/tianleiwu/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/tianleiwu/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/tianleiwu/subscriptions",
            "organizations_url": "https://api.github.com/users/tianleiwu/orgs",
            "repos_url": "https://api.github.com/users/tianleiwu/repos",
            "events_url": "https://api.github.com/users/tianleiwu/events{/privacy}",
            "received_events_url": "https://api.github.com/users/tianleiwu/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-06-18T06:53:51Z",
        "updated_at": "2020-06-18T06:53:51Z",
        "author_association": "MEMBER",
        "body": "Also, the NaN issue has been addressed:\r\nhttps://github.com/microsoft/onnxruntime/blob/2605faef88714b9e9d5ad0eac7f17c387db4cc1c/onnxruntime/contrib_ops/cuda/bert/attention_impl.cu#L158\r\n\r\nClose this issue now. Feel free to reopen it if the issue exists in correctly export model.",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/645818227/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/655324120",
        "html_url": "https://github.com/microsoft/onnxruntime/issues/3977#issuecomment-655324120",
        "issue_url": "https://api.github.com/repos/microsoft/onnxruntime/issues/3977",
        "id": 655324120,
        "node_id": "MDEyOklzc3VlQ29tbWVudDY1NTMyNDEyMA==",
        "user": {
            "login": "lintseju",
            "id": 2860147,
            "node_id": "MDQ6VXNlcjI4NjAxNDc=",
            "avatar_url": "https://avatars.githubusercontent.com/u/2860147?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/lintseju",
            "html_url": "https://github.com/lintseju",
            "followers_url": "https://api.github.com/users/lintseju/followers",
            "following_url": "https://api.github.com/users/lintseju/following{/other_user}",
            "gists_url": "https://api.github.com/users/lintseju/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/lintseju/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/lintseju/subscriptions",
            "organizations_url": "https://api.github.com/users/lintseju/orgs",
            "repos_url": "https://api.github.com/users/lintseju/repos",
            "events_url": "https://api.github.com/users/lintseju/events{/privacy}",
            "received_events_url": "https://api.github.com/users/lintseju/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2020-07-08T06:49:54Z",
        "updated_at": "2020-07-08T06:51:56Z",
        "author_association": "NONE",
        "body": "Hi @tianleiwu , I encounter the same problem, but I am using model from tf 1.x\r\nIt works fine before I do optimization, but after optimized, the program crashed with the same error message.\r\nHow can I fix this?\r\n\r\n```\r\nfrom onnx_model_bert import BertOptimizationOptions\r\noptimization_options = BertOptimizationOptions('bert_tf')\r\noptimization_options.enable_gelu_approximation = True\r\noptimized_model = optimizer.optimize_model(OUTPUT_DIR + \"bert.onnx\", model_type='bert_tf', num_heads=12, hidden_size=768, opt_level=0,\r\n                                           optimization_options=optimization_options,)\r\n# optimized_model.use_dynamic_axes()\r\noptimized_model.save_model_to_file(OUTPUT_DIR + \"bert_opt.onnx\")\r\n```\r\n\r\n```\r\nsess = ort.InferenceSession(\"../onnx_models/bert_opt.onnx\")\r\ndef _test_infer_onnx(sess, input_feature, _label_list):\r\n    data = {\"in_input_ids:0\": np.array([input_feature.input_ids], dtype=np.int64),\r\n          \"in_input_mask:0\": np.array([input_feature.input_mask], dtype=np.int64),\r\n          \"in_segment_ids:0\": np.array([input_feature.segment_ids], dtype=np.int64)}\r\n    ts0 = timeit.default_timer()\r\n    result = sess.run([\"loss/ArgMax:0\", \"loss/Max:0\"], data)\r\n    ts1 = timeit.default_timer()\r\n    return _label_list[result[0][0]], result[1][0], ts1 - ts0\r\npred_onnx, pred_proba_onnx, time_onnx = _test_infer_onnx(sess, input_feature, _label_list)\r\n```\r\n\r\nError log:\r\n```\r\nTraceback (most recent call last):\r\n  File \"serving.py\", line 137, in <module>\r\n    main()\r\n  File \"serving.py\", line 133, in main\r\n    sess.run([\"loss/ArgMax:0\", \"loss/Max:0\"], feed)\r\n  File \"/Users/xxx/anaconda3/lib/python3.7/site-packages/onnxruntime/capi/session.py\", line 111, in run\r\n    return self._sess.run(output_names, input_feed, run_options)\r\nonnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running SkipLayerNormalization node. Name:'SkipLayerNorm1' Status Message: skip is expected to have same shape as input\r\n```\r\n\r\nsystem info:\r\nGenerate ONNX model:\r\nGoogle colab\r\nONNX Runtime installed from (source or binary): pypi\r\nonnx==1.7.0\r\nonnxruntime==1.3.0\r\ntensorflow==1.15.2\r\ntf2onnx==1.6.2\r\nPython 3.6.9\r\n\r\nInference:\r\nMac OSX 10.15.5\r\nONNX Runtime installed from (source or binary): pypi\r\nonnx==1.7.0\r\nonnxruntime==1.3.0\r\ntensorflow==1.15.2\r\nPython 3.7.6",
        "reactions": {
            "url": "https://api.github.com/repos/microsoft/onnxruntime/issues/comments/655324120/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    }
]